lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1069,28 +1118,55 @@ class MongoClientInterface {
         });
     }
 
-    getObjectMDStats(bucketName, log, callback) {
+    getObjectMDStats(bucketName, bucketInfo, log, callback) {
         const c = this.getCollection(bucketName);
-        c.distinct('_id').then(keys => {
-            const trimmedKeys = keys.map(key => key.replace(/\0.*$/g, ''));
-            const uniqKeys = trimmedKeys.filter(
-                (key, index, self) => self.indexOf(key) === index);
-            // for each uniqKey get master version id
-            return async.map(uniqKeys, (key, done) => {
-                this.getLatestVersion(c, key, log, (err, mst) => {
+        const retBucketInfo = {
+            name: bucketName,
+            location: bucketInfo.getLocationConstraint(),
+        };
+        if (bucketInfo._versioningConfiguration &&
+            bucketInfo._versioningConfiguration.Status) {
```

Comment:
Can `Status` be truthy and not `Enabled`?

Reply:
`Status` can either be values `Suspended` or `Enabled`.